### PR TITLE
Re-enable automountServiceAccountToken for Kong

### LINKED ISSUE #628: Release 2.10.0 broke Linkerd injection
Hi Kong team,

I use Kong together with Linkerd service mesh. Unfortunately, the change of `automountServiceAccountToken` to `false`, which came with [PR #619](https://github.com/Kong/charts/pull/619) breaks Linkerd auto injection. Linkerd uses ServiceAccount for mTLS Identity verification purposes. I will prepare a PR to make `automountServiceAccountToken=false` and the corresponding volumemount configurable.

Kind regards,
Szymon

### PR DESCRIPTION
**Pull request by sgrzemski**
#### Which issue this PR fixes
  - fixes #628 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] PR is based off the current tip of the `main` branch.
- [X] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [X] New or modified sections of values.yaml are documented in the README.md
- [] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/kong/values.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ deployment:
   ## Specify the service account to create and to be assigned to the deployment / daemonset and for th migrations
   serviceAccount:
     create: true
+    automountServiceAccountToken: false
```
Comment (jrsmroz) [inline charts/kong/values.yaml:28]: Would you mind adding a comment describing the property?

Comment (sgrzemski) [reply charts/kong/values.yaml:28]: Added, thank you for pointing that out!

**Comment by jrsmroz** [comment]
@rainest Can you take a look? I wonder how does this PR impacts a security hardening that you've done.

File: charts/kong/values.yaml [hunk 2]
```diff
@@ -19,9 +19,12 @@ deployment:
     # Setting this to false with ingressController.enabled=true will create a
     # controller-only release.
     enabled: true
-  ## Specify the service account to create and to be assigned to the deployment / daemonset and for th migrations
+  ## Specify the service account to create and to be assigned to the deployment / daemonset and for the migrations
   serviceAccount:
     create: true
+    # The created service account can be automounted to the Kong deployment or
+    # it can be mounted as a regular volume
```
Comment (rainest) [inline charts/kong/values.yaml]:
```suggestion
    # Automount the service account token. By default, this is disabled, and the token is only mounted on the controller
    # container. Some sidecars require enabling this. Note that enabling this exposes Kubernetes credentials to Kong
    # Lua code, increasing potential attack surface.
```

File: charts/kong/Chart.yaml [hunk 3]
```diff
@@ -10,7 +10,7 @@ maintainers:
   email: traines@konghq.com
 name: kong
 sources:
-version: 2.10.2
+version: 2.10.3
```
Comment (rainest) [inline charts/kong/Chart.yaml]: Please strip this change, there are several PRs in flight at the moment and it'll be easier for us to release all their changes at once.

File: charts/kong/CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,7 @@
 ## Unreleased
 
 * Fix URL for more detail on Command Line Arguements / environment variables for Helm.
+* Make `automountServiceAccountToken` disabled but default, but configurable
```
Comment (rainest) [inline charts/kong/CHANGELOG.md]:
```suggestion
* Add `deployment.serviceAccount.automountServiceAccountToken` to force token mounting on.
```

**Review by rainest — CHANGES_REQUESTED** [review]

File: charts/kong/Chart.yaml (hunk 3 above)
Comment (rainest) [inline charts/kong/Chart.yaml]:
```suggestion
version: 2.10.2
```

**Review by rainest — DISMISSED** [review]

File: charts/kong/templates/deployment.yaml [hunk 5]
```diff
@@ -38,7 +38,7 @@ spec:
   template:
     metadata:
       annotations:
-        {{- if or .Values.deployment.serviceAccount.create .Values.deployment.serviceAccount.name }}
+        {{ - if (and (not .Values.deployment.serviceAccount.automountServiceAccountToken) (or .Values.deployment.serviceAccount.create .Values.deployment.serviceAccount.name)) }}
```
Comment (rainest) [inline charts/kong/templates/deployment.yaml]:
```suggestion
        {{- if (and (not .Values.deployment.serviceAccount.automountServiceAccountToken) (or .Values.deployment.serviceAccount.create .Values.deployment.serviceAccount.name)) }}
```

**Comment by rainest** [comment]
Ah, drat, CLA is missing from this, so even applying changes myself we can't merge this yet. @sgrzemski can you sign that? We'd like to include this in a release this week if possible.

**Comment by sgrzemski** [comment]
Hello @rainest,
Pardon my delay, I am on vacation and do not really reach out to my mail :D
Thank you for applying the suggestions! The CLA was agreed, but the commits had a wrong email. Rebased the branch and it looks good now. Sorry for the inconvenience caused.
Kind regards,
Szymon

**Review by rainest — APPROVED** [review]

**Pull request merged**